### PR TITLE
chore: test environment overrides on lb web svc manifest fields

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR is part of the first step (writing units test) to address #2492 by writing ApplyEnv unit tests on 
1. Fields that are exclusive to load-balanced web services (hence not covered by #2700)
2. All first-level fields for load-balanced web service

Some unit tests are marked "FAILED_TEST:.." and commented out. Those tests indicate potential bugs in our current manifest, and should be fixed in a later stage.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/manifest/lb_web_svc.go [hunk 1]
```diff
@@ -18,9 +18,8 @@ const (
 
 // Default values for HTTPHealthCheck for a load balanced web service.
 const (
-	DefaultHealthCheckPath                = "/"
-	DefaultHealthCheckDeregistrationDelay = 60
```
Comment (iamhopaul123) [inline internal/pkg/manifest/lb_web_svc.go:22]: Do we not use it 🤦

File: internal/pkg/manifest/lb_web_svc_apply_env_test.go [hunk 2]
```diff
@@ -0,0 +1,1514 @@
...
+		//	},
+		//	wanted: func(svc *LoadBalancedWebService) {
+		//		svc.HealthCheck = HealthCheckArgsOrString{
+		//			HealthCheckArgs: HTTPHealthCheckArgs{
+		//				Path: aws.String("/test"),
+		//			},
+		//		}
+		//	},
+		//},
+		//"FAILED_TEST: composite fields: args is overridden if str9jg is not nil": {
```
Comment (iamhopaul123) [inline internal/pkg/manifest/lb_web_svc_apply_env_test.go]:
```suggestion
		//"FAILED_TEST: composite fields: args is overridden if string is not nil": {
```

File: internal/pkg/manifest/lb_web_svc_apply_env_test.go [hunk 3]
```diff
@@ -0,0 +1,1514 @@
...
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Platform = &PlatformArgsOrString{
+					PlatformString: aws.String("mockPlatformTest"),
+				}
+			},
+		},
+		"platform not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Platform = &PlatformArgsOrString{
```
Comment (iamhopaul123) [inline internal/pkg/manifest/lb_web_svc_apply_env_test.go:1225]: Do we not need to test if `PlatformString` can override the args like https://github.com/aws/copilot-cli/pull/2728/files#diff-b30de9ebe3c4859ebd00dbc4695a04c836597aff05a1fdc25cb2cc923ff83037R279

File: internal/pkg/manifest/applyenv_test.go [hunk 4]
```diff
@@ -1103,6 +1103,10 @@ func TestApplyEnv_Image_HealthCheck(t *testing.T) {
 	}
 }
 
+func TestApplyEnv_Platform(t *testing.T) {
```
Comment (efekarakus) [inline internal/pkg/manifest/applyenv_test.go:1106]: Can we remove this 🙏 ?

File: internal/pkg/manifest/lb_web_svc.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/manifest/lb_web_svc.go:22]: We don't!

File: internal/pkg/manifest/lb_web_svc_apply_env_test.go (hunk 3 above)
Comment (Lou1415926) [reply internal/pkg/manifest/lb_web_svc_apply_env_test.go:1225]: We do! It's going to be added here: https://github.com/aws/copilot-cli/blob/ed91dbc861196fe5c156cba07043d79291f78d33/internal/pkg/manifest/applyenv_test.go#L1106

**Review by efekarakus — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**